src/components/Modal/__tests__/Modal.spec.tsx

```diff
@@ -0,0 +1,227 @@
+        if (onBeforeTest) {
+          await onBeforeTest(page);
+        }
+
+        const component = await mount(
+          <ModalForTest
+            {...props}
+          />,
+        );
+
+        if (onBeforeSnapshot) {
+          await onBeforeSnapshot(page, component);
+        }
+
+        const screenshot = await component.screenshot({ animations: 'disabled' });
+        expect(screenshot).toMatchSnapshot();
+      });
+    });
+  });
+
+  test.describe('non-visual', () => {
+    test('id', async ({ mount }) => {
+      const testId = 'testId';
+
+      const component = await mount(
+        <div>
+          <ModalForTest id={testId} />
+        </div>,
+      );
+
+      await expect(component.locator('dialog')).toHaveAttribute('id', testId);
+    });
+  });
+
+  test.describe('functionality', () => {
+    test('close on esc key press when allowCloseOnEscapeKey enabled', async ({ mount }) => {
+      let called = false;
+
+      const component = await mount(
+        <ModalForCallbackTest
+          allowCloseOnEscapeKey
+          closeButtonOnClick={() => {
+            called = true;
+          }}
+        />,
+      );
+
+      const dialog = component.locator('dialog');
+      await dialog.focus();
+      await dialog.press('Escape');
+      expect(called).toBe(true);
+    });
+
+    test('do not close on esc key press when allowCloseOnEscapeKey disabled', async ({ mount }) => {
+      let called = false;
+
+      const component = await mount(
+        <ModalForCallbackTest
+          allowCloseOnEscapeKey={false}
+          closeButtonOnClick={() => {
+            called = true;
+          }}
+        />,
+      );
+
+      const dialog = component.locator('dialog');
+      await dialog.focus();
+      await dialog.press('Escape');
+      expect(called).toBe(false);
+    });
+
+    test('call primary action on enter key press when input/select content focused', async ({ mount }) => {
+      let called = false;
+
+      const component = await mount(
+        <ModalForCallbackTest
+          allowPrimaryActionOnEnterKey
+          primaryButtonOnClick={() => {
+            called = true;
+          }}
+        />,
+      );
+
+      const input = component.locator('input[name="input1"]');
+      await input.focus();
+      await input.press('Enter');
+      expect(called).toBe(true);
+    });
+
+    test('do not call primary action on enter key press when allowPrimaryActionOnEnterKey is disabled', async ({ mount }) => {
+      let called = false;
+
+      const component = await mount(
+        <ModalForCallbackTest
+          allowPrimaryActionOnEnterKey={false}
+          primaryButtonOnClick={() => {
+            called = true;
+          }}
+        />,
+      );
+
+      const input = component.locator('input[name="input1"]');
+      await input.focus();
+      await input.press('Enter');
+      expect(called).toBe(false);
+    });
+
+    test('focus first input element when autoFocus enabled', async ({ mount }) => {
```

Comment:
Can you extend tests of autoFocus? First element is focused, then primary button and then any focusable element. See the docs for more. Can you read the docs and cover all the cases?

I am leaving following decision up to you, but due to large amount of test cases under `functionality` block, I would consider introduction of multiple describe block under `functionality` block by the name of `prop`.

E.g.:

`test.describe('allowCloseOnBackdropClick', ...)`
`test.describe('allowCloseOnEscapeKey', ...)`
`test.describe('allowPrimaryActionOnEnterKey', ...)`
`test.describe('allowCloseOnEscapeKey', ...)`
`test.describe('portalId', ...)`
`test.describe('id', ...)`

Speaking of that, I would like to have everywhere, but it is not part of this task. Leaving this up to you.